libs/execution_context/ebpf_core.c

```diff
@@ -1750,6 +1755,34 @@ _ebpf_core_get_pid_tgid()
     return ((uint64_t)ebpf_platform_process_id() << 32) | ebpf_platform_thread_id();
 }
 
+_Success_(return == 0) static uint32_t
+    _ebpf_core_get_current_logon_id(_In_ const void* ctx, _Out_ uint64_t* logon_id, int size)
+{
+    UNREFERENCED_PARAMETER(ctx);
+    UNREFERENCED_PARAMETER(size);
+
+    if (!ebpf_is_preemptible()) {
+        return 1;
+    }
+
+    ebpf_result_t result = ebpf_platform_get_authentication_id(logon_id);
+    if (result != EBPF_SUCCESS) {
+        return 1;
+    }
+
+    return 0;
+}
+
+_Success_(return == 0) static uint32_t _ebpf_core_is_current_admin(_In_ void* ctx, _Out_ uint32_t* is_admin, int size)
+{
+    // TODO: Issue# 1871 - Implement this function.
```

Comment:
The sock_addr specific implementation is implemented. Only the "global" or "default" implementation is pending. One of the reasons this API can fail is that if this is called at DISPATCH. Security APIs only allow PASSIVE_LEVEL. 

For example, since this is a global helper function, if an XDP program calls this helper function, it would typically be calling it at dispatch. So this API will need to return an error.

Reply:
Since each prog type decides whether to be dispatch-specific or passive-specific, should we put the distinction in something the verifier can verify?   For example, if this were a program-type-specific helper the verifier could verify it though that wouldn't scale to a bunch of program types that it is applicable to.   As is, this arguably encourages developers to not check the return value since "it always works", and then copying such code to another hook because the code calls a "global helper so it should work", which results in flawed code.

Reply:
Programs of a specific program type can be called at both PASSIVE and DISPATCH. For example, for `cgroup/connect*` attach types, the eBPF program can be called both at PASSIVE (for TCP) and at DISPATCH (for UDP).

Reply:
Some can yes.  Some can't.  My question is about helpers and program types that have a constraint.

Reply:
Agree that if we add such support in verifier, we can then provide this information for each helper to verifier.
But also, each such helper should be able to return an error (which we are doing for the new 2 helper functions added in this PR).

I can file a new issue to add such support.

Reply:
Filed #1900 to track this work.